File: code-playbook/spring-boot-4-querydsl-sql-demo_2/src/main/java/com/example/querydslsqldemo/entity/OrderEntity.java
package com.example.querydslsqldemo.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Entity
@Table(name = "orders")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class OrderEntity {

    @Id
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "ordered_at", nullable = false)
    private LocalDateTime orderedAt;

    @Column(name = "product_id", nullable = false)
    private Long productId;

    @Column(name = "amount", nullable = false, precision = 19, scale = 2)
    private BigDecimal amount;

    @Column(name = "status", nullable = false, length = 20)
    private String status;
}

File: practice_kotlin/kotlin-trading-core/src/main/kotlin/com/revy/trading/adapter/output/persistence/entity/OrderEntity.kt
package com.revy.trading.adapter.output.persistence.entity

import com.revy.trading.adapter.output.persistence.entity.common.VersionedEntity
import com.revy.trading.domain.enums.OrderSide
import com.revy.trading.domain.enums.OrderStatus
import jakarta.persistence.*
import java.math.BigDecimal
import java.util.*

@Entity
@Table(name = "orders")
class OrderEntity(
    @Column(name = "order_id", nullable = false, unique = true, updatable = false, columnDefinition = "uuid")
    val orderId: UUID,

    @Column(name = "account_id", nullable = false, updatable = false, columnDefinition = "uuid")
    val accountId: UUID,

    @Column(nullable = false, updatable = false, length = 20)
    val symbol: String,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, updatable = false, length = 10)
    val side: OrderSide,

    @Column(nullable = false, updatable = false)
    val quantity: Long,

    @Column(name = "limit_price", nullable = false, precision = 19, scale = 6, updatable = false)
    val limitPrice: BigDecimal,

    @Column(nullable = false, length = 3, updatable = false)
    val currency: String,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 30)
    var status: OrderStatus,

    @Column(name = "filled_quantity", nullable = false)
    var filledQuantity: Long,

    ) : VersionedEntity() {

}